src/Data/Food/Builder/Recipe.elm

```diff
@@ -163,6 +169,32 @@ compute db =
                             |> Maybe.map (computeProcessImpacts db.impacts >> List.singleton >> updateImpacts)
                             |> Maybe.withDefault Impact.noImpacts
 
+                    conservationImpacts =
+                        let
+                            wellknown =
+                                db.processes
+                                    |> Process.loadWellKnown
+
+                            mass =
+                                ingredients
+                                    |> List.map .mass
+                                    |> Quantity.sum
+
+                            volume =
+                                case ingredients of
+                                    [ i ] ->
+                                        i.mass |> Quantity.at_ (gramsPerCubicCentimeter i.ingredient.density)
+
+                                    _ ->
+                                        --density 1 for recipes
+                                        mass |> Quantity.at_ (gramsPerCubicCentimeter 1)
+                        in
+                        Result.map2 (Retail.computeImpacts db mass volume)
+                            wellknown
+                            (conservation
+                                |> Result.fromMaybe "No conservation defined"
+                            )
+
```

Comment:
This could be extracted to its own function outside of `compute`, what do you think?

Reply:
I've moved the mass and volume computation outside and reused the existing mass computation that takes the raw/cooked ratio into account.